src/main/java/com/google/sps/daos/DatabaseUserDao.java

```diff
@@ -108,6 +108,9 @@ private void initUser(String userId) {
   private void registerUserInGameInstance(DocumentReference userInGameInstanceDocRef, String userId) {
     Map<String, Object> docData = new HashMap<>();
     docData.put("points", 0);
+    docData.put("numberAnswered", 0);
+    docData.put("numberCorrect", 0);
+    docData.put("numberWrong", 0);
```

Comment:
You can have a function on **`GameInstance.java`** that generates this automatically, so you'll only need to call the function from an Game Instance object, which in this case it will be empty and you may not need it, but in other cases it's really helpful.
### **Example from the `Game.java` class**
```Java
  public Map<String, Object> gameData() {

    Map<String, Object> gameData = new HashMap<>();
        gameData.put("creator", this.creator());
        gameData.put("title", this.title());
        gameData.put("headQuestion", this.headQuestion());
        gameData.put("numberOfQuestions", this.questions().size());
    return gameData;
  }
```

Reply:
Mmm nice, I think we first need to do some work with `GameInstance.java` first to have all the info we need there? But this is something that definitely will come in handy when we refactor and abstract logic!

Reply:
+1 nice comment rodrigo!